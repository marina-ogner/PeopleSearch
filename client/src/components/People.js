import React from "react";
import Person from "./Person";
import Loader from 'react-loader';

class People extends React.Component {
    render() {
        if (this.props.people.length > 0) {
            var peopleAll = [];
            for (var i = 0; i < this.props.people.length; i++) {
                peopleAll.push(<Person inf={this.props.people[i]} num={i + 1} key={i + 1} />);
            }
        }
        return (
            <div>
                {this.props.load && <div><Loader 
                    loaded={false} lines={13} length={20} width={10} radius={20}
                    corners={1} rotate={0} direction={1} color="#000" speed={1}
                    trail={60} shadow={false} hwaccel={false} className="spinner"
                    zIndex={2e9} top="50%" left="50%" scale={1.00}
                    loadedClassName="loadedContent" /></div>}
                {this.props.people.length > 0 && <div className="containerMy">{peopleAll}</div>}
                {this.props.error && <div className="alert alert-danger fade in alertMy" role="alert">
                    {this.props.error}
                </div>}
            </div>

        );
    }
};

export default People;