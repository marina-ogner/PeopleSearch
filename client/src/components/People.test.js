import React from 'react';
import People from './People';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

test('People component should render as expected', () => {
    const people = [["MARK", "PRENTICE", "2413 W TEXAS ST", "MOSES LAKE", "WA"], ["MARK", "PRENTICE", "11107 50TH AVE SE", "EVERETT", "WA"]];
    const component = shallow(<People people={people} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(2);
    expect(tree).toMatchSnapshot();
});
test('People component should render as expected with only one matching person', () => {
    const people = [["MARK", "PRENTICE", "2413 W TEXAS ST", "MOSES LAKE", "WA"]];
    const component = shallow(<People people={people} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(1);
    expect(tree).toMatchSnapshot();
});
test('People component should render as expected without data about people (with this.prop.people=[])', () => {
    const people = [];
    const component = shallow(<People people={people} />);
    const tree = toJson(component);
    expect(tree.children).toBe(null);
    expect(tree).toMatchSnapshot();
});
test('People component should render as expected when the data is not fetching (with this.prop.people=false)', () => {
    const people = [];
    const load = false;
    const component = shallow(<People people={people} load={load} />);
    const tree = toJson(component);
    expect(tree.children).toBe(null);
    expect(tree).toMatchSnapshot();
});
test('People component should render as expected when the data is fetching (with this.prop.people=true)', () => {
    const people = [];
    const load = true;
    const component = shallow(<People people={people} load={load} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(1);
    expect(tree).toMatchSnapshot();
});
test('People component should render as expected with error (with this.prop.error)', () => {
    const people = [];
    const error = "something is wrong";
    const component = shallow(<People people={people} error={error} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(1);
    expect(tree).toMatchSnapshot();
});
test('People component should render as expected without error (!this.prop.error)', () => {
    const people = [];
    const error = undefined;
    const component = shallow(<People people={people} error={error} />);
    const tree = toJson(component);
    expect(tree.children).toBe(null);
    expect(tree).toMatchSnapshot();
});
