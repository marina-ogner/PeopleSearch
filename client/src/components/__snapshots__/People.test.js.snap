// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People component should render as expected 1`] = `
<div>
  <div
    className="containerMy"
  >
    <Person
      inf={
        Array [
          "MARK",
          "PRENTICE",
          "2413 W TEXAS ST",
          "MOSES LAKE",
          "WA",
        ]
      }
      key="1"
      num={1}
    />
    <Person
      inf={
        Array [
          "MARK",
          "PRENTICE",
          "11107 50TH AVE SE",
          "EVERETT",
          "WA",
        ]
      }
      key="2"
      num={2}
    />
  </div>
</div>
`;

exports[`People component should render as expected when the data is fetching (with this.prop.people=true) 1`] = `
<div>
  <div>
    <Component
      className="spinner"
      color="#000"
      component="div"
      corners={1}
      direction={1}
      hwaccel={false}
      left="50%"
      length={20}
      lines={13}
      loaded={false}
      loadedClassName="loadedContent"
      parentClassName="loader"
      radius={20}
      rotate={0}
      scale={1}
      shadow={false}
      speed={1}
      top="50%"
      trail={60}
      width={10}
      zIndex={2000000000}
    />
  </div>
</div>
`;

exports[`People component should render as expected when the data is not fetching (with this.prop.people=false) 1`] = `<div />`;

exports[`People component should render as expected with error (with this.prop.error) 1`] = `
<div>
  <div
    className="alert alert-danger fade in alertMy"
    role="alert"
  >
    something is wrong
  </div>
</div>
`;

exports[`People component should render as expected with only one matching person 1`] = `
<div>
  <div
    className="containerMy"
  >
    <Person
      inf={
        Array [
          "MARK",
          "PRENTICE",
          "2413 W TEXAS ST",
          "MOSES LAKE",
          "WA",
        ]
      }
      key="1"
      num={1}
    />
  </div>
</div>
`;

exports[`People component should render as expected without data about people (with this.prop.people=[]) 1`] = `<div />`;

exports[`People component should render as expected without error (!this.prop.error) 1`] = `<div />`;
